### PR TITLE
{AKS} vendor 2021-10-01 API

### PR DESCRIPTION
**Pull request by tyler-lloyd**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/vendored_sdks/azure_mgmt_preview_aks/_container_service_client.py [hunk 1]
```diff
@@ -121,6 +121,7 @@ def models(cls, api_version=DEFAULT_API_VERSION):
            * 2021-07-01: :mod:`v2021_07_01.models<azure.mgmt.containerservice.v2021_07_01.models>`
            * 2021-08-01: :mod:`v2021_08_01.models<azure.mgmt.containerservice.v2021_08_01.models>`
            * 2021-09-01: :mod:`v2021_09_01.models<azure.mgmt.containerservice.v2021_09_01.models>`
+           * 2021-10-01: :mod:`v2021_10_01.models<azure.mgmt.containerservice.v2021_10_01.models>`
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/vendored_sdks/azure_mgmt_preview_aks/_container_service_client.py:124]: Please modify the 59th line and change it to `DEFAULT_API_VERSION = '2021-10-01'`. You can copy this file directly from the SDK.

**Review by FumingZhang — COMMENTED** [review]
I highly recommend that you only vendor 2021-10-01 api in this PR and take another PR to add new features.

Besides, you can refer this [wiki](https://dev.azure.com/msazure/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/177284/Upgrade-SDK-and-API-Version-for-acs-and-aks-preview) to vendor a new api version.

**Review by FumingZhang — APPROVED** [review]
The PR looks good to me :shipit:

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**